### PR TITLE
perf: Cases page

### PR DESCRIPTION
**Pull request by jfaracik**
Skeleton loading

* Overhauls the `/cases/` page using JavaScript - drastically faster, sub one second to load the page with 15k cases and 1.5k flags on the system
* New expand control for flags, animates flags in five at a time
* New assigned users control
* SLA has been rewritten to be more consistent

The **vast** majority of files changed are due to moving BasePage from `shared` into `pages`.

### CODE REVIEW

File: assets/javascripts/cases.js [hunk 1]
```diff
@@ -1,29 +1,190 @@
...
-	if ($(".lite-expand__control").length == $(".lite-expand__control--open").length) {
-		$("#link-open-all-table-folds").addClass("lite-expand__base-control--open");
+function getSLADaysRingColour(remainingDays) {
+    if (remainingDays > 5) {
+		return "green"
+	} else if (remainingDays >= 0) {
+		return "orange"
 	} else {
-		$("#link-open-all-table-folds").removeClass("lite-expand__base-control--open");
+		return "red"
```
Comment (iichr) [inline assets/javascripts/cases.js:31]: can that be a negative number then (overdue) ?

File: pipeline/Jenkinsfile [hunk 2]
```diff
@@ -70,10 +70,10 @@ parameters
                     sh "py.test -k '${env.TESTS_TO_RUN}' -n '${env.NO_OF_PARALLEL_RUNNERS}' --dist=loadfile --ignore=core --disable-pytest-warnings --alluredir=ui_automation_tests/allure-results"
                 } else {
                     sh """
-                    py.test -k 'smoke' --reruns 1 --ignore=core --disable-pytest-warnings --alluredir=ui_automation_tests/allure-results
+                    py.test -k 'smoke' --reruns '${env.NO_OF_RERUNS}' --ignore=core --disable-pytest-warnings --alluredir=ui_automation_tests/allure-results
                     if [ \$? -eq 0 ]
                     then
-                         py.test -k 'regression' -n '${env.NO_OF_PARALLEL_RUNNERS}' --dist=loadfile --reruns 2 --ignore=core --disable-pytest-warnings --alluredir=ui_automation_tests/allure-results
+                         py.test -k 'regression' -n '${env.NO_OF_PARALLEL_RUNNERS}' --dist=loadfile --reruns '${env.NO_OF_RERUNS}' --ignore=core --disable-pytest-warnings --alluredir=ui_automation_tests/allure-results
```
Comment (iichr) [inline pipeline/Jenkinsfile:76]: 👌

File: templates/queues/cases.html [hunk 3]
```diff
@@ -50,195 +50,210 @@ <h1 class="govuk-heading-l">
...
+							<p>hello world my name</p>
+						</div>
+						<div class="">
+							<p>hello is word</p>
+						</div>
+						<div class="">
+							<p>easy</p>
+						</div>
+						<div class="">
+							<p>to find</p>
```
Comment (iichr) [inline templates/queues/cases.html:98]: easy to find indeed 👀 rm

File: assets/javascripts/cases.js (hunk 1 above)
Comment (jfaracik) [reply assets/javascripts/cases.js:31]: Yep!

File: templates/queues/cases.html (hunk 3 above)
Comment (jfaracik) [reply templates/queues/cases.html]: Fixed oops :P

**Pull request merged**